### PR TITLE
Matchers support

### PR DESCRIPTION
**Pull request by kassenov**
Matchers support, at the moment for 3.0.x but going to make another for master branch.

### CODE REVIEW

**Comment by leleuj** [comment]
Looks good to me. A few comments:

1) Can you target the 3.1.x branch I created? If it's too complicated, keep things that way

2) Can you upgrade the version to 3.1.x? (in the pom.xml file)

3) Can you update the README with the new versions and the new capabilities (please use the same text as available in the https://github.com/pac4j/j2e-pac4j project?

4) Can you also update to pac4j v2.1 by the way (you won't have no change)?

5) Can you update the demos to test that new feature?

**Comment by kassenov** [comment]
On it. For demo projects, I understand that I should wait for this PR to be merged. Is it right?

File: README.md [hunk 1]
```diff
@@ -381,7 +389,7 @@ Rules for the security filter can be supplied in application.conf. An example is
 consists of a list of filter rules, where the key is a regular expression that will be used to
 match the url. Make sure that the / is escaped by \\ to make a valid regular expression.
 
-For each regex key, there are two subkeys: `authorizers` and `clients`. Here you can define the
+For each regex key, there are two subkeys: `authorizers`, `clients` and `matchers`. Here you can define the
```
Comment (leleuj) [inline README.md]: "three" in fact. I'll change that by myself on the README after the merge...

**Comment by leleuj** [comment]
Really excellent! Thanks for your contribution. I'm merging the pull request so you will be able to easily update the demos...

**Comment by leleuj** [comment]
And don't forget the PR for the master...

**Comment by kassenov** [comment]
Few things:
1. It seems that the new version should be in maven in order to be able to use it in the demos
2. We need to fix versions and small bits of other text in Readme.md
3. I will create a PR for master after updating demos, if it's ok.

**Comment by leleuj** [comment]
You're right, I fixed the README and the build: https://oss.sonatype.org/content/repositories/snapshots/org/pac4j/play-pac4j/3.1.0-SNAPSHOT/

You can now update the demos, I created the 3.1.x branch for that on both demos.

Then, you'll need to do the same again for the master branch (both on the library itself and on the demos)...

**Pull request merged**